src/commands/roll.js

```diff
@@ -1,85 +1,85 @@
+}, { result: 0, rolls: [] })
+  
+const reduceRolls = (a, b) => {
+    const regEx = /[+-](\d+)d(\d+|f)/
+    const exp = b.match(regEx)
+
+    const isFudge = exp[2] == "f" ? true : false
+    const roll = isFudge ? roller(exp[1], -1, 1) : roller(exp[1], 1, exp[2])
+    const rollArr = isFudge ? a.rolls.concat([roll.rolls.map(x => fudgify(x))]) : a.rolls.concat([roll.rolls])
 
-function numberToFudge(number) {
-    if (number > 0) return '+';
-    if (number < 0) return '-';
-    if (number == 0) return '  ';
+    return exp[0].startsWith("+") 
+        ? { result: a.result + roll.result, rolls: rollArr }
+        : { result: a.result - roll.result, rolls: rollArr }
 }
 
-function roll (text) {
-    try {
-        const [ _, n, dice ] = text.match(rollRegEx);
-        const rolls = new Array;
-        let total = 0;
-        let modifier =addModifier(text);
-        modifier = parseInt(modifier);
-        for (let i=n;i>=1;i--) {
-            let result;
-            if (dice == 'f') {
-                result = Math.floor(Math.random() * 3) -1 ;
-            }else{
-                result = Math.floor(Math.random() * dice) + 1;
-            }
-            total += result;
-            if (dice == 'f') result = numberToFudge(result);
-            rolls.push(result);
-        }
-        total += modifier;
-        let result = total;
-        if (n > 1) result += ' (' + rolls.join() + ')';
-        return result;
-    }catch (e) {
-        return (defaultRoll 
-            ? roll(defaultRoll + text)
-            : 'You must set a default roll or give a valid expression.'
-        );
-    }
+const reduceNumber = (a, b) => a + parseInt(b)
+
+const addModifiers = (text, regEx, reducer, initVal) => {
+    const mod = text.match(regEx)
+    return mod != null ? mod.reduce(reducer, initVal) : 0
 }
 
-function checkDefault() {
-    return 'Default roll is ' + defaultRoll + '.';
+const addNumbers = text => {
+    const regEx = /[+-]\d+(?!d(?:\d+|f))/g
+    return addModifiers(text, regEx, reduceNumber, 0)
 }
 
-function setDefault (newDefault, message) {
-    if (checkRole(message.author.id, message.guild.roles)) {
-        try {
-            const [ roll, n, dice ] = rollRegEx.exec(newDefault);
-            defaultRoll = roll;
-            return 'Default roll has been set to ' + defaultRoll + '.';
-        }catch (e) {
-            return 'You must insert a valid expression.';
-        }
-    }else{
-        return 'Sorry, you don\'t have permission to do that.';
-    }
+const addRolls = text => { 
+    const regEx = /[+-](\d+)d(\d+|f)/g
+    return addModifiers(text, regEx, reduceRolls, { result: 0, rolls: [] })
 }
 
-module.exports.setDefaultRoll = (text, message) => {
-    if (text.toLowerCase().startsWith('set default roll')) return message.channel.send(setDefault(text, message));
+const parseRolls = text => {
+    const splitter = /(?=[^+-]\d+d(?:\d+|f))/
+    const trimmer = /[^-+0-9df]/g
```

Comment:
Not that it matters, but these regexes can be declared at root level of the module so that they won't be recreated every time the function is invoked.

Reply:
I suppose I could get all regExes out of the functions.

Reply:
Yeah, that wouldn't be bad.